text/0089-tenant-token.md

```diff
@@ -0,0 +1,125 @@
+- Title: Tenant Tokens
+- Start Date: 2021-10-15
+- Specification PR: [#89](https://github.com/meilisearch/specifications/pull/89)
+- Discovery Issue: [#51](https://github.com/meilisearch/product/issues/51)
+
+# Tenant Token
+
+## 1. Functional Specification
+
+### I. Summary
+
+The SDKs can generate `Tenant tokens` inheriting from a MeiliSearch's `API key` generated to force filters during a search for an end-user. A `Tenant token` is generated on the user server-side code to be used by an end-user. It allows users to have multi-tenant indexes and thus restricts access to documents depending on the end-user making the search request.
+
+### II. Motivation
+
+`Tenant tokens` are introduced to solve multi-tenant use-cases. By providing the management of API Keys, we remove one of the last major deal-breakers that makes users not choose MeiliSearch as a solution despite all our advantages.
+
+Users regularly request Multi-Tenant indexes. Users today need to set up workarounds to meet this need. Some of them implement reverse-proxy or managed authentication systems like Hasura or Kong to filter what can and cannot be read. Others decide to use server code as a facade to implement the access restriction logic before requesting MeiliSearch. It is difficult to maintain, less efficient, and requires necessary skills that not everyone has.
+
+### III. Glossary
+
+| Term               | Definition |
+|--------------------|------------|
+| Master Key         | This is the master key that allows the creation of other API keys. The master key is defined by the user when launching MeiliSearch. |
+| API Key            | API keys are stored and managed from the endpoint `/keys` by the master key holder. These are the keys used by the technical teams to interact with MeiliSearch at the level of the frontend/backend code. |
+| Tenant Token     | These tokens are not stored and managed by a MeiliSearch instance. They are generated for each end-user by the backend code from a MeiliSearch API Key. They are used by the end-users to only search the documents belonging to them. |
+| Multi-Tenancy      | By multi-tenancy, we mean that an end-user only accesses data belonging to him within an index shared with other end-users. |
```

Comment:
I'm missing a definition of what we mean by tenant. Is every end-user a tenant?

Reply:
I've added a definition in the specifcation. Let me know if it's clearer!